src/controller/userRouter.js

```diff
@@ -1,12 +1,12 @@
 const express = require('express');
 const userService = require('../services/userService');
-const { validateUsername, validatePassword } = require('../middleware/userMiddleware');
+const { validateUsername, validatePassword, validateRole } = require('../middleware/userMiddleware');
 const {  authenticate, adminAuthenticate  } = require("../middleware/authMiddleware");
 const { handleServiceError } = require("../utilities/routerUtilities");
 
 const userRouter = express.Router();
 
-userRouter.post("/", validateUsername, validatePassword, async (req, res) => {
+userRouter.post("/", validateUsername(), validatePassword(), async (req, res) => {
```

Comment:
This may break middleware. Unless the function calls are returning callback functions.

Reply:
That is exactly what is happening, I reworked `userMiddleware` and `postMiddleware` to reduce the boilerplate code, and as a result they now return callback functions.